pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -77,7 +77,7 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedDeploymentState(ctx *workflow.
 	}
 }
 
-func advancedDeploymentIdle(ctx *workflow.Context, project *akov2.AtlasProject, deployment *akov2.AtlasDeployment, atlasDeploymentAsAtlas *mongodbatlas.AdvancedCluster) (*mongodbatlas.AdvancedCluster, workflow.Result) {
+func advancedDeploymentIdle(ctx *workflow.Context, k8sClient client.Client, project *akov2.AtlasProject, deployment *akov2.AtlasDeployment, atlasDeploymentAsAtlas *mongodbatlas.AdvancedCluster) (*mongodbatlas.AdvancedCluster, workflow.Result) {
```

Comment:
I wonder if `ctx *workflow.Context` did not already reference the k8s client.

Not sure if that is the best way, but we should be consistent and not pass it separately on some places and embedded in others.